Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs

```diff
@@ -32,12 +33,18 @@ public static bool AttemptAnswer(Riddle riddle, RiddleAnswer answer)
         // TODO: Make this punish the player for an incorrect answer
         // TODO: Potentially properly support "select all correct answers" questions depending on if the client wants them
         Debug.Log($"Attempted answer to riddle {riddle.Question} with answer {answer.Answer} and result {riddle.CorrectAnswers.Contains(answer)}");
-        return riddle.CorrectAnswers.Contains(answer);
+        bool result = riddle.CorrectAnswers.Contains(answer);
+        if (result && solved_riddles != null)
+        {
+            solved_riddles.Add(riddle.Question);
+        }
+        return result;
     }
 
     private static void LoadRiddlesFromXML(string filePathToRiddlesFolder)
     {
         riddles = new List<Riddle>();
+        solved_riddles = new HashSet<string>();
```

Comment:
I think we should initialize this HashSet where it's declared so we don't have to worry about it being null before this method gets called

Reply:
okay! let me change that